src/components/interactive-builder/add-question-modal.component.tsx

```diff
@@ -675,6 +709,80 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
+                      <ComboBox
+                        id="programLookup"
+                        items={programs}
+                        itemToString={(item: Program) => item?.name}
+                        onChange={({ selectedItem }: { selectedItem: Program }) => {
+                          handleProgramChange(selectedItem);
+                        }}
+                        placeholder={t('addProgram', 'Add program')}
+                        selectedItem={selectedProgram}
+                        titleText={t('program', 'Program')}
+                      />
+                    )}
+
+                    {selectedProgram && (
+                      <ComboBox
+                        id="programWorkflowLookup"
+                        items={programWorkflows}
+                        itemToString={(item: ProgramWorkflow) => item?.concept?.display}
+                        onChange={({ selectedItem }: { selectedItem: ProgramWorkflow }) =>
+                          handleProgramWorkflowChange(selectedItem)
+                        }
+                        placeholder={t('addProgramWorkflow', 'Add program workflow')}
+                        selectedItem={programWorkflowUuid}
+                        titleText={t('programWorkflow', 'Program workflow')}
+                      />
+                    )}
+                  </div>
+                  {programWorkflowUuid && (
+                    <div>
+                      {isLoadingProgramStates && <SelectSkeleton />}
+                      {programStatesLookupError && (
+                        <InlineNotification
+                          kind="error"
+                          lowContrast
+                          className={styles.error}
+                          title={t('errorFetchingProgramState', 'Error fetching program state')}
+                          subtitle={t('pleaseTryAgain', 'Please try again.')}
+                        />
+                      )}
+                      {programStates?.length > 0 && (
+                        <MultiSelect
+                          titleText={t('programState', 'Program state')}
+                          id="programState"
+                          items={programStates}
+                          itemToString={(item: ProgramState) => (item ? item?.concept?.display : '')}
+                          selectionFeedback="top-after-reopen"
+                          onChange={(data: ProgramStateData) => setSelectedProgramState(data.selectedItems)}
+                          selectedItems={selectedProgramState}
+                        />
+                      )}
+                      {selectedProgramState?.map((answer) => (
+                        <Tag className={styles.tag} key={answer?.uuid} type={'blue'}>
+                          {answer?.concept?.display}
+                        </Tag>
+                      ))}
+                    </div>
+                  )}
+                </Stack>
+              )}
```

Comment:
This whole bit should move into the `<Stack>` too. Also, we shouldn't need the parent div.